src/schema.ts

```diff
@@ -43,15 +46,30 @@ export class Schema {
         }
       } else if (primitiveType === PrimitiveType.DATE) {
         type = Type.TEMPORAL;
+        // need to get correct min/max of date data
+        summary.min = new Date(data[0][field]);
+        summary.max = new Date(data[0][field]);
+        for (var i = 0; i < data.length; i++) {
+          const time = new Date(data[i][field]).getTime();
+          if (time < (summary.min as Date).getTime()) {
+            summary.min = new Date(time);
+          }
+          if (time > (summary.max as Date).getTime()) {
+            summary.max = new Date(time);
+          }
+        }
       } else {
         type = Type.NOMINAL;
       }
-      return {
+      var fieldSchema: FieldSchema = {
```

Comment:
Why declaring this variable?

1) It seems like you're not recalculating any binStats at all although calculating the default bin would be helpful? 
2) I just realize I was wrong about not recalculating cardinalities for time units at all.  
We should pre-calculate them to support #22.